### PR TITLE
Bump rubocop version to 1.48.1.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Also fix all of the new warnings.

[b/272577695](http://b/272577695).

See also https://github.com/GoogleCloudPlatform/fluent-plugin-detect-exceptions/pull/101.

### CODE REVIEW

File: fluent-plugin-google-cloud.gemspec [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
   # https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/config/software/fluentd.rb.
   gem.add_runtime_dependency 'fluentd', '1.13.3'
+  gem.add_runtime_dependency 'google-api-client', '0.30.8'
   gem.add_runtime_dependency 'googleapis-common-protos', '1.3.10'
   gem.add_runtime_dependency 'googleauth', '0.9.0'
-  gem.add_runtime_dependency 'google-api-client', '0.30.8'
   gem.add_runtime_dependency 'google-cloud-logging', '1.6.6'
   gem.add_runtime_dependency 'google-protobuf', '3.20.0'
   gem.add_runtime_dependency 'grpc', '1.45.0'
-  gem.add_runtime_dependency 'json', '2.2.0'
```
Comment (LujieDuan) [inline fluent-plugin-google-cloud.gemspec:32]: Just want to check: is this intended upgrade?

**Review by LujieDuan — APPROVED** [review]
LGTM! 

Thanks for having each fix in its own commit - that makes verifying each enforcement very easy. 

The presubmit is failed as `rubocop=1.48.1` requires ruby >= 2.6. Should we modify the test matrix in this PR?

File: fluent-plugin-google-cloud.gemspec (hunk 1 above)
Comment (igorpeshansky) [reply fluent-plugin-google-cloud.gemspec:32]: Yes, I believe this was required by the newer version of Rubocop…

**Review by LujieDuan — APPROVED** [review]
LGTM!

**Pull request merged**